### PR TITLE
Account Activity enhancements

### PR DESCRIPTION
**Pull request by pandasoft0**
Fix issues #745 
Continue to work on tasks started at PR #755, thanks to @eleete0712 

- disable pull to refresh at fragments and add it at the top of activity
- fix all issues mentioned at #755 

Don't know how to solve the issue:
- mark suspended profiles and profiles that have blocked the current user
Need help with this issue

Add a loading placeholder to the top of the view looks a bit messed at the timeline fragment, swipe to refresh is disabled. So, i added an indeterminate progress bar to the top of timeline and medias fragments to show data is refreshing.

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java [hunk 1]
```diff
@@ -180,19 +184,21 @@ public StatusViewData apply(Either<Placeholder, Status> input) {
...
         TimelineFragment fragment = new TimelineFragment();
         Bundle arguments = new Bundle();
         arguments.putString(KIND_ARG, kind.name());
+        arguments.putBoolean(ARG_ENABLE_SWIPE_TO_REFRESH,enableSwipeToRefresh);
         fragment.setArguments(arguments);
         return fragment;
     }
 
-    public static TimelineFragment newInstance(Kind kind, String hashtagOrId) {
+    public static TimelineFragment newInstance(Kind kind, String hashtagOrId, boolean enableSwipeToRefresh) {
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java]: maybe add more overloads for the `newInstance` methods, so `enableSwipeToRefresh` only has to be set when its actually false?

```
public static TimelineFragment newInstance(Kind kind, String hashtagOrId) {
   return  newInstance(kind, hashtagOrId, false);
}

```

File: app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt [hunk 2]
```diff
@@ -96,7 +94,7 @@ class AccountActivity : BottomSheetActivity(), ActionButtonActivity, HasSupportF
     @ColorInt
     private var textColorSecondary: Int = 0
     @Px
-    private var avatarSize: Float = 0f
+    private var avatarSize: Int = 0
```
[the remark this answers is not in the document]
Comment (pandasoft0) [reply app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt]: This field annotated with @Px annotation:
`Denotes that an integer parameter, field or method return value is expected` 
So, we should remove Px annotation and change the type to Int back (and replace `getDimensionPixelSize` with `getDimension`) or leave it as is. What is you prefer?

File: app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt [hunk 3]
```diff
@@ -144,7 +142,15 @@ class AccountActivity : BottomSheetActivity(), ActionButtonActivity, HasSupportF
         setContentView(R.layout.activity_account)
 
         val intent = intent
-        accountId = intent.getStringExtra(KEY_ACCOUNT_ID)
+        val accountId = intent.getStringExtra(KEY_ACCOUNT_ID)
+        val activeAccount = accountManager.activeAccount
+
+        // Obtain information to fill out the profile.
+        viewModel.setAccountInfo(accountId,accountId == activeAccount?.accountId)
```
[the remark this answers is not in the document]
Comment (pandasoft0) [reply app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt]: Good point, thanks!

File: app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt [hunk 4]
```diff
@@ -32,10 +32,15 @@ class AccountViewModel  @Inject constructor(
...
 
+    val isRefreshing = MutableLiveData<Boolean>().apply { value=false }
+    private var isDataLoading = false
 
-    fun obtainAccount(accountId: String, reload: Boolean = false) {
-        if(accountData.value == null || reload) {
+    lateinit var accountId:String
+    var isSelf = false
 
+    fun obtainAccount(reload: Boolean = false) {
```
[the remark this answers is not in the document]
Comment (pandasoft0) [reply app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt]: Done

File: app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java (hunk 1 above)
Comment (pandasoft0) [reply app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java]: Done

File: app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt]: Ah I see. I think removing the annotation is better.

[the remark this answers is not in the document]
Comment (pandasoft0) [reply app/src/main/java/com/keylesspalace/tusky/AccountActivity.kt]: Done

File: app/src/main/res/layout/activity_account.xml [hunk 5]
```diff
@@ -1,316 +1,355 @@
...
+                        tools:text="Follow Requested" />
 
-                <ViewStub
-                    android:id="@+id/accountMovedView"
-                    android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:inflatedId="@+id/accountMovedView"
-                    android:layout="@layout/view_account_moved"
-                    app:layout_constraintTop_toBottomOf="@id/accountFieldList" />
+                    <ImageButton
```
[the remark this answers is not in the document]
Comment (pandasoft0) [reply app/src/main/res/layout/activity_account.xml]: I replaced ImageButton with MaterialButton, but set Start/End padding to 8dp. Default paddings are 16dp and button looks too wide.

**Comment by pandasoft0** [comment]
> * there is a crash when reloading moved profiles
> * Probably the Posts/Follows/Followers count should also be shown on moved profiles, so the only difference is the additional info.
Hm. i followed your account https://mastodon.social/@ConnyDuck but it is looks like a normal account and i can't check these issues. I need to have moved account to check this and next issues :( Have you ideas how to do the account moved?

> * I think the mute button should only be visible when the user actually is muted.
So, show the button only when user muted and hide it if user clicks unmute. So, should work as unmute button only. Correct?

> * the loading indicator of the pull-to-refresh needs to be styled in Tusky colors, see the other fragment on how it is done
Done

> * please format your code, I see many instances of missing spaces etc that make it hard to read
I did auto format for all files i edited.

**Comment by pandasoft0** [comment]
Refactored code a bit. Split very long `onCreate` and `onAccountChanged` methods on some short methods. Rebased code to latest main branch (to change Picasso with Glide)

**Comment by connyduck** [comment]
Ah yeah I switched connyduck@mastosoc back, sorry, now its moved again.

> I replaced ImageButton with MaterialButton, but set Start/End padding to 8dp. Default paddings are 16dp and button looks too wide.

nice

> So, show the button only when user muted and hide it if user clicks unmute. So, should work as unmute button only. Correct?

yes

**Comment by pandasoft0** [comment]
Thanks for your help.
- Moved account refresh crash is fixed
- Show stat for moved account (I've placed them above the "moved" view)
- Update "Mute" button behaviour

There is an issue with the "moved" account. Tabs and viewpager are hidden, but we can collapse the toolbar. I see the simple solution - disable collapse for "moved" accounts and hard solution - try to update the toolbar behaviour to stop collapse if bottom scroll view absent. I don't know how many time it will take. I have not enough experience to work with Collapse toolbars and scroll behaviours.

**Comment by connyduck** [comment]
just show the tabs on moved profiles? I think I proposed this in #745

**Comment by pandasoft0** [comment]
Sorry, i didn't understand you. Is thought you mean follows/followers count widgets. I will check it today.

**Comment by connyduck** [comment]
Yeah the follows/follower can stay as well. In fact, the only difference should be the moved notice shown.

**Comment by pandasoft0** [comment]
Done

**Comment by pandasoft0** [comment]
Sorry, fixed.

**Pull request merged**